client/state/plans/selectors/get-plan-prices.ts

```diff
@@ -15,6 +19,8 @@ export interface PlanPrices {
 /**
  * A convenience function that returns the undiscounted and discounted prices
  * for a given plan and site.
+ * If the planSlug passed is the current plan, then this function returns
+ * { rawPrice: sitePlanRawPrice, discountedRawPrice: 0, planDiscountedRawPrice: 0 }
```

Comment:
Thanks @jdc91, that makes sense to me and I agree with you. I think the root cause is that the billing timeframe and plan price are rendered by different components, with each component having it's own (duplicated) logic.

-----

<img width="549" alt="image" src="https://user-images.githubusercontent.com/5436027/236173468-9b83a267-15dc-480b-ac81-4508ba583ba2.png">

-----

How do we establish a guarantee that both components will display the same price? One solution is to have a common parent component for both these, which calculates the price to be shown and pass it down to the billing-timeframe and plan-price components as props. This would stop the duplication of logic and also ensure that the prices shown are consistent between both components. WDYT?

Reply:
Ah yes! That is a valid concern and your solutions sounds just about right! 💯